### PR TITLE
Adds a 404 handler for Django which responds with JSON

### PR DESCRIPTION
**Pull request by bmbouter**
https://pulp.plan.io/issues/1140
closes #1140

### CODE REVIEW

**Comment by bmbouter** [comment]
ok test

File: server/pulp/server/webservices/views/util.py [hunk 1]
```diff
@@ -111,3 +111,25 @@ def wrapper(*args, **kwargs):
...
+    have additional parameters.
+
+    :param request: The WSGI object containing the request
+    :type request: django.core.handlers.wsgi.WSGIRequest
+    :param args: Additional positional arguments
+    :type args: list
+    :param kwargs: Additional keyword arguments
+    :type kwargs: dict
+
+    :return: A response object that contains a 404 status code and an empoty json payload.
```
Comment (asmacdo) [inline server/pulp/server/webservices/views/util.py]: s/empoty/empty/

**Comment by asmacdo** [comment]
@bmbouter, nice work, thanks!

File: server/pulp/server/webservices/views/util.py [hunk 2]
```diff
@@ -111,3 +111,25 @@ def wrapper(*args, **kwargs):
...
+    :rtype: django.http.HttpResponse
+    """
+    json_response = generate_json_response()
+    json_response.status_code = 404
```
Comment (mhrivnak) [inline server/pulp/server/webservices/views/util.py]: It's not very important, but I personally like using the constants from httplib. In this case, ``httplib.NOT_FOUND``.

Comment (bmbouter) [inline server/pulp/server/webservices/views/util.py]: Good idea. I updated this line, the rest of this module, and the associated tests to use `httplib`.

**Pull request merged**